### DISCUSSION TITLE
Multicast Failures

### DISCUSSION THREAD

**Discussion by Botched1** [zwave-js/zwave-js#7194] [General] [answered]
I originally posted this in the other thread, but my issue sounds like it may be different than his so I thought I would start my own thread.

I'm using a ZST39 running 7.22.1 and see multicast commands fail somewhat regularly. If I repeatedly send multicast commands I can get it to give the error maybe 3 out of 10 times or so... 

I'm running:
zwave-js-ui: 9.20.0
zwave-js: 13.3.1

Integrated to home assistant via the web socket route (not MQTT).

What I see in the logs when it fails:

```
2024-09-25 20:30:38.430 ERROR Z-WAVE-SERVER: The node did not acknowledge the command (ZW0204)
ZWaveError: The node did not acknowledge the command (ZW0204)
    at Driver.sendMessage (/opt/node_modules/zwave-js/src/lib/driver/Driver.ts:5816:23)
    at Driver.sendCommandInternal (/opt/node_modules/zwave-js/src/lib/driver/Driver.ts:6014:28)
    at Driver.sendCommand (/opt/node_modules/zwave-js/src/lib/driver/Driver.ts:6124:29)
    at BinarySwitchCCAPI.set (/opt/node_modules/@zwave-js/cc/src/cc/BinarySwitchCC.ts:142:24)
    at BinarySwitchCCAPI.<anonymous> (/opt/node_modules/@zwave-js/cc/src/cc/BinarySwitchCC.ts:164:16)
    at VirtualNode.setValue (/opt/node_modules/zwave-js/src/lib/node/VirtualNode.ts:172:39)
    at MulticastGroupMessageHandler.handle (/opt/node_modules/@zwave-js/server/dist/lib/multicast_group/message_handler.js:19:50)
    at Client.receiveMessage (/opt/node_modules/@zwave-js/server/dist/lib/server.js:139:100)
    at WebSocket.<anonymous> (/opt/node_modules/@zwave-js/server/dist/lib/server.js:60:45)
    at WebSocket.emit (node:events:517:28)
```

In my case I send the command from node-red to Home Assistant. It is configured as follows to turn a group of 4 switches off:

command: zwave_js.multicast_set_value

```
{
    "entity_id": [
        "switch.parlor_front_left_light",
        "switch.parlor_front_right_light",
        "switch.parlor_rear_left_light",
        "switch.parlor_rear_right_light"
    ],
    "command_class": 37,
    "endpoint": 0,
    "property": "targetValue",
    "property_key": 1,
    "value": false
}
```

**Comment by AlCalzone** [comment]
Please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs) showing the issue, loglevel `debug` and attach it here as a file (drag & drop into the text field).

**Comment by Botched1** [comment]
Here is what I learned...

The last multicast command that issued an error was at 18:22:49.

* I don't see any obvious errors or issues in the driver log (attached).
* I do see an error just like the one in the OP in the DEBUG log as viewed in the zwavejs ui.

Jason
[zwave-js-ui-store.zip](https://github.com/user-attachments/files/17157087/zwave-js-ui-store.zip)

**Reply by AlCalzone** [reply]
That's not a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs). That's the application log.

**Reply by Botched1** [reply]
Yep, I'm an idiot. 

I'll get the right log this evening. Sorry about that.

**Reply by Botched1** [reply]
> That's not a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs). That's the application log.

Here you go. Failure is at 2024-09-27T12:33:26.205Z
[zwavejs_2024-09-27.log](https://github.com/user-attachments/files/17164206/zwavejs_2024-09-27.log)

**Reply by AlCalzone** [reply]
Hm that looks an awful lot like the issue in the other thread.

**Reply by AlCalzone** [reply, marked as answer]
Let's continue over [there](https://github.com/zwave-js/node-zwave-js/discussions/7184).

**Reply by Botched1** [reply]
Ah, maybe it is the same thing then.

I've tried rebuilding the routes to the 4 devices, no change. I'll go look over in that other thread and see if there are other things I should try.

I do have a UZB3 flashed as a zniffer, but it isn't hooked up. I can try to do that this weekend.

**Reply by AlCalzone** [reply]
You can do that. I'll see if I can reproduce myself, and report all findings to Silabs.
